src/components/Calculator.js

```diff
@@ -1,77 +1,102 @@
-      (state) => ({
-        dataObj: calculate(state.dataObj, buttonValue),
-      }),
-      function () {
-        let CalculatorScreen;
+  useEffect(() => {
+    let calculatorScreen;
+
+    if (dataObj.next == null && dataObj.total == null) {
+      calculatorScreen = 0;
+    } else if (dataObj.next) {
+      calculatorScreen = dataObj.next;
+    } else {
+      calculatorScreen = dataObj.total;
+    }
 
-        if (this.state.dataObj.next == null && this.state.dataObj.total == null) {
-          CalculatorScreen = 0;
-        } else if (this.state.dataObj.next) {
-          CalculatorScreen = this.state.dataObj.next;
-        } else {
-          CalculatorScreen = this.state.dataObj.total;
-        }
+    setCalculatorScreen(calculatorScreen);
+  }, [dataObj]);
 
-        this.setState({
-          CalculatorScreen,
-        });
-      },
-    );
+  const handleClick = (event) => {
+    const buttonValue = event.target.innerText;
+    setDataObj(calculate(dataObj, buttonValue));
   };
 
-  render() {
-    return (
-      <div className="calculator-container">
-        <input type="text" value={this.state.CalculatorScreen}></input>
-        <table>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>AC</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>+/-</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>%</td>
-            <td className="bg-orange" onClick={this.handleClick}>÷</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>7</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>8</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>9</td>
-            <td className="bg-orange" onClick={this.handleClick}>x</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>4</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>5</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>6</td>
-            <td className="bg-orange" onClick={this.handleClick}>-</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" onClick={this.handleClick}>1</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>2</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>3</td>
-            <td className="bg-orange" onClick={this.handleClick}>+</td>
-          </tr>
-          <tr>
-            <td className="bg-light-gray" colSpan="2" onClick={this.handleClick}>0</td>
-            <td className="bg-light-gray" onClick={this.handleClick}>.</td>
-            <td className="bg-orange" onClick={this.handleClick}>=</td>
-          </tr>
-        </table>
-      </div>
-    );
-  }
+  return (
+    <div className="calculator-container">
+      <input type="text" value={calculatorScreen}></input>
+      <table>
+        <tr>
```

Comment:
- [ ] If you look at your console you will see a complaint about this table structure. Basically, you need to follow the classic hierarchy of a table. Here is an example.

```
<table>
    <thead>
        <tr>
            <th>The table header</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>The table body</td>
            <td>with two columns</td>
        </tr>
    </tbody>
</table>

```
![Screenshot from 2021-09-25 11-32-02](https://user-images.githubusercontent.com/40503704/134768374-d3052a00-a5a6-4b31-8ddb-d54d69c86aaf.png)